art17/factsheet.py

```diff
@@ -410,6 +420,31 @@ def get(self):
                        view_func=HabitatHeader.as_view('habitat-header'))
 
 
+def generate_factsheet_url(category, subject, period):
+    if category == 'species':
+        model_cls = SpeciesMixin.model_cls
+        fs_cls = SpeciesFactSheet
+    elif category == 'habitat':
+        model_cls = HabitatMixin.model_cls
+        fs_cls = HabitatFactSheet
+    else:
+        raise NotImplementedError('Unknown category:', category)
+
+    period = period or app.config['FACTSHEET_DEFAULT_PERIOD']
+    assessment = (
+        model_cls.query.filter_by(subject=subject,
+                                  dataset_id=period).first_or_404()
+    )
+    pdf_path = str(
+        path(app.config['PDF_DESTINATION'] )
+        / fs_cls.get_pdf_file_name(assessment)
+    ) + '.pdf'
+    real_path = str(path(app.static_folder) / pdf_path)
+    if os.path.exists(real_path):
```

Comment:
`path.Path` instances have `exists()` method to check its presence. This way you get rid of the `os.path` import.
    
    real_path = path(app.static_folder) / pdf_path
    if real_path.exists()